exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -207,6 +211,19 @@ func (prwe *PrwExporter) handleHistogramMetric(tsMap map[string]*prompb.TimeSeri
 	return nil
 }
 
+// handleSummaryMetric processes data points in a single OTLP summary metric by mapping the sum, count and each
+// quantile of every data point as a Sample, and adding each Sample to its corresponding TimeSeries.
+// tsMap and metric cannot be nil.
+func (prwe *PrwExporter) handleSummaryMetric(tsMap map[string]*prompb.TimeSeries, metric *otlp.Metric) error {
+	if metric.GetDoubleSummary().GetDataPoints() == nil {
```

Comment:
This exact condition has already been checked in `validateMetrics()` so maybe unnecessary here. But I do see that the rest of the `handle` functions do the same check again so its probably fine

Reply:
it might be worth considering removing this check across all functions to be honest because it is just duplicate logic

Reply:
I think the validateMetrics condition is a little bit different. There's a check for if `GetDoubleSummary` is null and then if `.GetDoubleSummary().GetDataPoints()` is null. I think I'll keep it in for consistency as the check doesn't take long anyways.